src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs

```diff
@@ -308,6 +389,22 @@ bool ValidateInstructions(ReadOnlySpan<byte> code, in EofHeader header)
                     return false;
                 }
             }
+
+            foreach (int rjumpdest in rjumpdests)
+            {
+                foreach (var range in immediates)
+                {
+                    if (range.Includes(rjumpdest))
+                    {
+                        if (Logger.IsTrace)
+                        {
+                            Logger.Trace($"EIP-4200 : Static Relative Jump destination {rjumpdest} is an Invalid, falls within {range}");
+                        }
+                        return false;
+                    }
+                }
+            }
```

Comment:
Aren't we missing checks that we aren't jumping to push data?

Reply:
we are doing it here  : https://github.com/NethermindEth/nethermind/blob/05d716a362b2db958fc32c11b8ebe10628fa0431/src/Nethermind/Nethermind.Evm/EOF/EvmObjectFormat.cs#L379

Reply:
Yeah I'm stupid :)

One thing though we were optimizing this analysis in `CodeDataAnalyzer` and heuristic to use `CodeDataAnalyzer` or `JumpdestAnalyzer`, I think this might be optimized here?

Reply:
I am not sure I get what you are implying but these EOF doesn't allow dynamic jumps so ``CodeDataAnalyzer`` and ``JumpdestAnalyzer``  are never used in EOF context, unless you mean "get inspired" by how it was optimized over there to do the same here

Reply:
What I mean that in `CodeDataAnalyzer` we create a bitmap of invalid jumps. Creating this bitmap was heavily optimized. Now question is would using this approach instead of `immediates` for both PUSH and jump offsets would be faster?

It might be as we would go down from O(n^2) to O(n) here, while keeping O(n) creation time.

Reply:
well here is a simple approach (still trying to understand the bitmap approach tho) https://github.com/NethermindEth/nethermind/pull/5101, also does this method take allocations into consideration ?

Reply:
i have a different approach as well, but i could not test it. let me know if it can be optimized @LukaszRozmej 
```
var list = rjumpdests.ToList();
foreach (var range in immediates)
{
    var index = list.BinarySearch(range.Start.Value);
    if (index >= 0)
    {
        if (Logger.IsTrace)
        {
            Logger.Trace($"EIP-4200 : Static Relative Jump destination {list[index]} is an Invalid, falls within {range}");
        }
        return false;
    }
    else if (~index != list.Count && list[~index] < range.End.Value)
    {
        if (Logger.IsTrace)
        {
            Logger.Trace($"EIP-4200 : Static Relative Jump destination {list[~index]} is an Invalid, falls within {range}");
        }
        return false;
    }
}
```